app/controllers/SummarySubscriptionsController.scala

```diff
@@ -36,17 +38,39 @@ class SummarySubscriptionsController @Inject()(
                                                 requireData: DataRequiredAction,
                                                 val controllerComponents: MessagesControllerComponents,
                                                 view: SummarySubscriptionsView,
-                                                navigator: Navigator
+                                                navigator: Navigator,
+                                                sessionRepository: SessionRepository
                                               )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
-  def onPageLoad(mode: Mode, year: String, index: Int): Action[AnyContent] = (identify andThen getData andThen requireData) {
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
     implicit request =>
 
-      request.userAnswers.get(TaxYearSelectionPage) match {
-        case Some(taxYears) =>
-          Ok(view(taxYears, navigator.nextPage(SummarySubscriptionsPage, mode, request.userAnswers).url))
+      (
+        request.userAnswers.get(TaxYearSelectionPage),
+        request.userAnswers.get(SummarySubscriptionsPage),
+        request.userAnswers.get(NpsData)
+      ) match {
+        case (Some(taxYears), None, Some(npsData)) =>
+          val subs = taxYears.flatMap(
+            taxYear =>
+              Map(getTaxYear(taxYear) -> SummaryData(Seq.empty, npsData(getTaxYear(taxYear).toString)))
+          ).toMap
+
+          Ok(view(subs, navigator.nextPage(SummarySubscriptionsPage, mode, request.userAnswers).url, mode))
+        case (Some(taxYears), Some(subscriptions), Some(npsData)) =>
+          val subs = taxYears.flatMap(
+            taxYear => {
```

Comment:
Can we create val for result of getTaxYear(taxYear) rather than call the method 6 times